RecoParticleFlow/PFClusterProducer/plugins/LegacyPFClusterProducer.cc

```diff
@@ -0,0 +1,251 @@
+          vDefaults[1].addParameter<std::string>("detector", "HCAL_ENDCAP");
+          vDefaults[1].addParameter<std::vector<int>>("depths", {1, 2, 3, 4, 5, 6, 7});
+          vDefaults[1].addParameter<std::vector<double>>("logWeightDenominator", {0.1, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2});
+          bar.addVPSet("logWeightDenominatorByDetector", validator, vDefaults);
+        }
+        bar.add<double>("minAllowedNormalization", 1e-9);
+        pfClusterBuilder.add("allCellsPositionCalc", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        bar.add<double>("corrTermLowE", 0.);
+        bar.add<double>("threshLowE", 6.);
+        bar.add<double>("noiseTerm", 21.86);
+        bar.add<double>("constantTermLowE", 4.24);
+        bar.add<double>("noiseTermLowE", 8.);
+        bar.add<double>("threshHighE", 15.);
+        bar.add<double>("constantTerm", 2.82);
+        pfClusterBuilder.add("timeResolutionCalcBarrel", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        bar.add<double>("corrTermLowE", 0.);
+        bar.add<double>("threshLowE", 6.);
+        bar.add<double>("noiseTerm", 21.86);
+        bar.add<double>("constantTermLowE", 4.24);
+        bar.add<double>("noiseTermLowE", 8.);
+        bar.add<double>("threshHighE", 15.);
+        bar.add<double>("constantTerm", 2.82);
+        pfClusterBuilder.add("timeResolutionCalcEndcap", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        pfClusterBuilder.add("positionReCalc", bar);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        pfClusterBuilder.add("energyCorrector", bar);
+      }
+      desc.add("pfClusterBuilder", pfClusterBuilder);
+    }
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  // the actual algorithm
+  std::unique_ptr<PFCPositionCalculatorBase> _positionCalc;
+  std::unique_ptr<PFCPositionCalculatorBase> _allCellsPositionCalc;
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  const edm::EDGetTokenT<reco::PFClusterHostCollection> alpakaPfClustersToken;
+  const edm::EDGetTokenT<reco::PFRecHitFractionHostCollection> alpakaPfRHFractionToken;
+  const edm::EDGetTokenT<reco::PFRecHitHostCollection> InputPFRecHitSoA_Token_;
+  const edm::ESGetToken<reco::PFClusterParamsHostCollection, JobConfigurationGPURecord> pfClusParamsToken;
+  const edm::EDPutTokenT<reco::PFClusterCollection> legacyPfClustersToken;
+  const edm::EDGetTokenT<reco::PFRecHitCollection> recHitsLabel;
+
+  int nRH = 0;
+};
+
+void LegacyPFClusterProducer::produce(edm::Event& event, const edm::EventSetup& setup) {
+  const reco::PFRecHitHostCollection& pfRecHits = event.get(InputPFRecHitSoA_Token_);
+
+  reco::PFClusterHostCollection const& clusterSoA = event.get(alpakaPfClustersToken);
+  reco::PFRecHitFractionHostCollection const& fractionSoA = event.get(alpakaPfRHFractionToken);
+
+  auto const& alpakaPfClusters = clusterSoA.const_view();
+  auto const& alpakaPfRhFrac = fractionSoA.const_view();
```

Comment:
By the time we get to this module, these two actually don't have anything to do with alpaka.
Could you rename to `pfClusterSoA` or `pfClusterView` or something like that ?

Or even, combine these with the `event.get` into
```c++
auto const& pfClusterSoA = event.get(alpakaPfClustersToken).const_view();
```
?